var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -289,17 +294,36 @@ def determine_b2_options(self, spec, options):
                 'toolset=%s' % self.determine_toolset(spec)
             ])
 
+        cxxflags = []
+        if spec.variants['cppstandard'].value == '98':
+            cxxflags.append(self.compiler.cxx98_flag)
+        elif spec.variants['cppstandard'].value == '11':
+            cxxflags.append(self.compiler.cxx11_flag)
+        elif spec.variants['cppstandard'].value == '14':
+            cxxflags.append(self.compiler.cxx14_flag)
+        elif spec.variants['cppstandard'].value == '17':
+            cxxflags.append(self.compiler.cxx17_flag)
+        elif spec.variants['cppstandard'].value == 'default':
```

Comment:
I think you don't need the code from here down to the end of the `INTERNAL ERROR`. Variants should be already checked when parsing specs (i.e. the `tty.die` part will never be reached), and if you just `pass` there's no need for an extra condition.

Reply:
The `else:` clause is not there as a check on the validity of the user's input: it's there as a protection against the spec growing a new allowable value for `cppstandard` which the future author forgets to actually account for where it matters (it happens -- ask me how I know). Conversely, the `pass` is there as a statement that the author is aware that the value, `default` exists (hence its categorization as an internal error rather than user configuration), and that the correct thing to way to account for it is to do nothing. I can put comments in to that effect to forestall future questions, but unless there's a coding convention against this flavor of defensive programming, I'd rather leave the code in place.